From 0616f679f6d6c17a41d57623e822b60082a13696 Mon Sep 17 00:00:00 2001
From: Manjunath A Kumatagi <mkumatag@in.ibm.com>
Date: Fri, 19 Aug 2016 12:05:46 -0700
Subject: [PATCH] Add ppc64le kernel path

---
 livemedia-creator | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/livemedia-creator b/livemedia-creator
index f698bde..0cf69b6 100755
--- a/livemedia-creator
+++ b/livemedia-creator
@@ -204,9 +204,8 @@ class IsoMountpoint(object):
         else:
             self.mount_dir = self.initrd_path
 
-        self.kernel = self.mount_dir+"/isolinux/vmlinuz"
-        self.initrd = self.mount_dir+"/isolinux/initrd.img"
-
+        kernel_list = [("/isolinux/vmlinuz", "/isolinux/initrd.img"),
+                       ("/ppc/ppc64/vmlinuz", "/ppc/ppc64/initrd.img")]
         if os.path.isdir( self.mount_dir+"/repodata" ):
             self.repo = self.mount_dir
         else:
@@ -214,9 +213,15 @@ class IsoMountpoint(object):
         self.liveos = os.path.isdir( self.mount_dir+"/LiveOS" )
 
         try:
-            for f in [self.kernel, self.initrd]:
-                if not os.path.isfile(f):
-                    raise Exception("Missing file on iso: {0}".format(f))
+            for kernel, initrd in kernel_list:
+                if (os.path.isfile(self.mount_dir+kernel) and
+                    os.path.isfile(self.mount_dir+initrd)):
+                    self.kernel = self.mount_dir+kernel
+                    self.initrd = self.mount_dir+initrd
+                    break
+            else:
+                raise Exception("Missing kernel and initrd file in iso, failed"
+                                " to search under: {0}".format(kernel_list))
         except:
             self.umount()
             raise
-- 
1.8.3.1

